src/AWS.Deploy.CLI/Commands/ServerModeCommand.cs

```diff
@@ -59,33 +59,29 @@ public async Task ExecuteAsync(CancellationToken cancellationToken = default(Can
             }
             else
             {
-                var monitorTask = WaitOnParentPid(cancellationToken);
-                var webTask = host.RunAsync(cancellationToken);
-
-                //This call will wait until one of the tasks completes.
-                //The monitor task will complete if a parent pid is not found.
-                await Task.WhenAny(monitorTask, webTask);
-
-                //The monitor task is expected to complete only when a parent pid is not found.
-                if (monitorTask.IsCompleted && monitorTask.Result)
+                try
                 {
-                    _interactiveService.WriteLine(string.Empty);
-                    _interactiveService.WriteLine("The parent process is no longer running.");
-                    _interactiveService.WriteLine("Server mode is shutting down...");
-                    await host.StopAsync(cancellationToken);
+                    var process = Process.GetProcessById((int)_parentPid);
+                    process.EnableRaisingEvents = true;
+                    process.Exited += async (sender, args) => { await ShutDownHost(host, cancellationToken); };
                 }
-
-                //If the web task completes with a fault because an exception was thrown,
-                //We need to capture the inner exception and rethrow it so it can bubble up to the end user.
-                if (webTask.IsCompleted && webTask.IsFaulted)
+                catch (Exception)
                 {
-                    var innerException = webTask.Exception?.InnerException;
-                    if (innerException != null)
-                        throw innerException;
+                    return;
                 }
+
+                await host.RunAsync(cancellationToken);
```

Comment:
It appears like a regression, previously when the host task (web app) exited, we were throwing an exception which is no longer true now.

I think `ServerModeSession` will kick in and re-start the server but we are losing the diagnostics here IMO.

Reply:
Before we were using Task.WhenAny which causes an AggregateException to be thrown. We were catching that and throwing back the inner exception. Now we don't have that anymore and await host.RunAsync(cancellationToken) will just throw an exception same as before. The same behavior was retained.